tools/kmodules/lisa-in-tree/fetch_lisa_module.py

```diff
@@ -0,0 +1,84 @@
+#! /usr/bin/env python3
+
+#
+# Usage:
+#
+# Method 1 - LISA_HOME
+#
+# If the script finds the LISA_HOME environment variable (present after sourcing the Lisa shell)
+# it will symlink the Lisa module sources from the local Lisa git checkout into the kernel tree.
+# That way updating the local Lisa tree will automatically update the module sources in the kernel
+# tree and the changes will be present after the next kernel rebuild.
+# In this case, the script only needs to be run once.
+#
+# Method 2 - Sparse checkout
+#
+# If the script doesn't find LISA_HOME (e.g. building on an external server without a Lisa checkout)
+# the alternative mode is to directly clone the module sources into the tree. Assuming the paths are
+# set correctly and the remote repo exists:
+# $ ./fetch_lisa_module.py
+# will checkout and symlink the repo accordingly.
+# After the first checkout, re-running the script with a specific git hash will
+# run 'git checkout'.
+# $ ./fetch_lisa_module.py <HASH>
+# It is also possible to pass '-u' in order to fetch the latest sources prior
+# to the checkout.
+# $ ./fetch_lisa_module.py -u <HASH>
+
+import os
+import sys
+import argparse
+import subprocess
+
+# module directory setup
+LISA_MODULE_DIR = "lisa"
+LISA_MODULE_DIR_KERNEL_PATH = "private/gs-google/drivers/soc/arm/vh/kernel"
+
+# sparse clone remote setup
+LISA_MODULE_REMOTE = "git@github.com:mrkajetanp/lisa.git"
```

Comment:
This should be turned into an argparse option, defaulting to the upstream LISA repo